runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -188,29 +190,30 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
             }
         }
 
-        // Linearize this read against a timestamp
-        final long timestamp =
-                rt.getSequencerView()
-                        .nextToken(Collections.singleton(streamID), 0).getToken()
-                        .getTokenValue();
-        log.debug("Access[{}] conflictObj={} version={}", this, conflictObject, timestamp);
-
         // Perform underlying access
-        try {
-            return underlyingObject.access(o -> o.getVersionUnsafe() >= timestamp
-                            && !o.isOptimisticallyModifiedUnsafe(),
-                    o -> o.syncObjectUnsafe(timestamp),
-                    o -> accessMethod.access(o));
-        } catch (TrimmedException te) {
-            log.warn("Access[{}] Encountered Trim, reset and retry", this);
-            // We encountered a TRIM during sync, reset the object
-            underlyingObject.update(o -> {
-                o.resetUnsafe();
-                return null;
-            });
-            // And attempt an access again.
-            return accessInner(accessMethod, conflictObject, isMetricsEnabled);
+        for (int x = 0; x < rt.getParameters().getTrimRetry(); x++) {
+            // Linearize this read against a timestamp
+            final long timestamp = rt.getSequencerView()
+                            .nextToken(Collections.singleton(streamID), 0).getToken().getTokenValue();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "streamID"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)